oauth2_provider/oauth2_backends.py

```diff
@@ -122,7 +122,7 @@ def create_authorization_response(self, request, scopes, credentials, allow):
             credentials["user"] = request.user
 
             headers, body, status = self.server.create_authorization_response(
-                uri=credentials["redirect_uri"], scopes=scopes, credentials=credentials
+                uri=request.get_raw_uri(), scopes=scopes, credentials=credentials
```

Comment:
I don't understand why this changes from using the supplied `redirect_uri`. Does this allow one to send in a redirect to other than the approved place(s)?

Can you explain what's happening here?

Reply:
(I don't think anything actually uses this uri.)

The URI is passed to `oauthlib.oauth2.rfc6749.endpoints.AuthorizationEndpoint`, where it is used to construct an `oauthlib.common.Request`. I believe it is supposed to be the URI of the incoming request, and so it was an error for it to be passing the redirect uri as it was previously.

Its not documented as such in `AuthorizationEndpoint`, but if you look at the examples in `oauthlib.oauth2.rfc6749.grant_types.AuthorizationCodeGrant`, its clear that `oauthlib` is expecting the uri passed in to be the uri requesting the authorization endpoint:

https://github.com/oauthlib/oauthlib/blob/b69fa53fd836dc559aa7fcd78ce075bcbe361629/oauthlib/oauth2/rfc6749/grant_types/authorization_code.py#L210-L235

The redirect uri is also in `credentials`, which are all set as attributes on the `oauthlib.common.Request`. 

https://github.com/oauthlib/oauthlib/blob/b69fa53fd836dc559aa7fcd78ce075bcbe361629/oauthlib/oauth2/rfc6749/endpoints/authorization.py#L89-L95

Checking the redirect uri uses this `request.redirect_uri` attribute, not the request URI. 

https://github.com/oauthlib/oauthlib/blob/b69fa53fd836dc559aa7fcd78ce075bcbe361629/oauthlib/oauth2/rfc6749/grant_types/base.py#L228-L241